### ISSUE TITLE
[BUG] dataset.export() errors if label_cls is a list and label_fields is empty

### ISSUE THREAD

**Issue by laserK3000** [voxel51/fiftyone#2292]
### System information

-   **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: archlinux
-   **FiftyOne installed from (pip or source)**: pip
-   **FiftyOne version (run `fiftyone --version`)**:  FiftyOne v0.17.2, Voxel51, Inc.
-   **Python version**: Python 3.10.8

### Describe the problem

I am getting an error if the `label_cls` of our custom dataset exporter is a list rather than a tuple. According to the documentation it can be either a list, a tuple or a dict. The error only appears when the `label_fields` parameter of the `dataset.export` method is empty.

https://github.com/voxel51/fiftyone/blob/4e64ee7a6713cb799b6043bf8f90ec23fe092102/fiftyone/core/collections.py#L9166

IMHO either the documentation should be changed or the list of classes should be cast into one of the allowed types  of `issubclass`. I can provide a fix for both.

### Other info / logs
```
  File "/home/USER/filepath/tests/test_tfds.py", line 107, in <module>
    test_tfds_export()
  File "/home/USER/filepath/tests/test_tfds.py", line 88, in test_tfds_export
    dataset.export(
  File "/home/USER/filepath/.venv/lib/python3.10/site-packages/fiftyone/core/collections.py", line 6963, in export
    _export(
  File "/home/USER/filepath/.venv/lib/python3.10/site-packages/fiftyone/core/collections.py", line 9163, in _export
    label_field = sample_collection._parse_label_field(
  File "/home/USER/filepath/.venv/lib/python3.10/site-packages/fiftyone/core/collections.py", line 8205, in _parse_label_field
    return _parse_label_field(
  File "/home/USER/filepath/.venv/lib/python3.10/site-packages/fiftyone/core/collections.py", line 8451, in _parse_label_field
    label_field = _get_default_label_fields_for_exporter(
  File "/home/USER/filepath/.venv/lib/python3.10/site-packages/fiftyone/core/collections.py", line 8552, in _get_default_label_fields_for_exporter
    label_field_or_dict = _get_fields_with_types(
  File "/home/USER/filepath/.venv/lib/python3.10/site-packages/fiftyone/core/collections.py", line 8616, in _get_fields_with_types
    return _get_field_with_type(
  File "/home/USER/filepath/.venv/lib/python3.10/site-packages/fiftyone/core/collections.py", line 8642, in _get_field_with_type
    field = _get_matching_label_field(label_schema, label_cls)
  File "/home/USER/filepath/.venv/lib/python3.10/site-packages/fiftyone/core/collections.py", line 8694, in _get_matching_label_field
    if issubclass(field_type.document_type, label_type_or_types):
TypeError: issubclass() arg 2 must be a class, a tuple of classes, or a union
```

### What areas of FiftyOne does this bug affect?

-   [ ] `App`: FiftyOne application issue
-   [x] `Core`: Core `fiftyone` Python library issue
-   [ ] `Server`: Fiftyone server issue

### Willingness to contribute

The FiftyOne Community encourages bug fix contributions. Would you or another
member of your organization be willing to contribute a fix for this bug to the
FiftyOne codebase?

-   [ ] Yes. I can contribute a fix for this bug independently.
-   [x] Yes. I would be willing to contribute a fix for this bug with guidance
        from the FiftyOne community.
-   [ ] No. I cannot contribute a bug fix at this time.

**Comment by brimoor** [comment]
Good catch! Looks like list just needs to be cast to tuple in `fiftyone.core.collections. _get_matching_label_field()`. A PR would be most welcome if you're interested 🤗

**Comment by brimoor** [comment]
I went ahead and patched this in #2311
